2021/day14/day14.java

```diff
@@ -0,0 +1,88 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.io.File;
+
+public class day14 {
+    public static class Rule {
+        String pair;
+        String outcome;
+    }
+    // Read polymer template e.g. NCNBCHB
+    public static String readTemplate(String filename) {
+        String template = null;
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            template = sc.next();
+        } catch (Exception e) {
+            System.out.println("Something went wrong" + e);
+        } finally {
+            sc.close();
+        }
+        return template;
+    }
+    // Read list of rules e.g. CH -> B
```

Comment:
This method could use a description as there is some kind of transformation happening, but the current comment does not capture this. I would say something like

```java
/*
 * Given the name of a file containing a list of rules for creating
 * new elements in the form of arrow-delimited strings, e.g. "CH -> B",
 * read and transform them into a list of `Rule`s
 */
```

Reply:
Ok, I am just going to use that. There is no way I am going to come up with a better description